environs/networking.go

```diff
@@ -105,3 +117,19 @@ func SupportsContainerAddresses(env Environ) bool {
 	}
 	return ok
 }
+
+// ProviderSpaceInfo contains all the information about a space needed
+// by another environ to decide whether it can be routed to.
+type ProviderSpaceInfo struct {
+	network.SpaceInfo
+
+	// This identifies the cloud containing the space. Care must be
```

Comment:
I don't really have a strong opinion about this. In the implementation it's more convenient to work with fields that have a defined type rather than pulling everything from a `map[string]interface{}` and do checking for presence and type. It seems like every cloud is going to care about type, name, region and credential, with endpoint being useful for non-built-in clouds. You're right that identity and storage endpoints are very unlikely to be useful, though.

I can also see that having the structure is kind of inviting the client code to look at it (although I guess it could be private).